libs/ledger-live-common/src/families/solana/tx-fees.ts

```diff
@@ -14,12 +17,18 @@ export async function estimateTxFee(
   const tx = createDummyTx(address, kind);
   const [onChainTx] = await buildTransactionWithAPI(address, tx, api);
 
-  const fee = await api.getFeeForMessage(onChainTx.message);
+  let fee = await api.getFeeForMessage(onChainTx.message);
+  let retries = 0;
+
+  while (typeof fee !== "number" && retries++ < MAX_RETRIES) {
+    // FIXME HACK: sometimes fee is not a number, retrying with a next blockhash
+    log("debug", `api.getFeeForMessage returned invalid fee: <${fee}>`);
+    fee = await retryWithNewBlockhash(api, onChainTx);
```

Comment:
i would advice to add a small delay after this, it's a general good practice to add a delay (even one that becomes higher and higher) to avoid high pressure from client spamming an api

Reply:
You're right but it's already done in the internal implem of `retryWithNewBlockhash`
https://github.com/LedgerHQ/ledger-live/blob/26f589097ccdeb5853aba2e2ce963533287bb681/libs/ledger-live-common/src/families/solana/tx-fees.ts#L186